### PR TITLE
Update variables, cleanup, remove `{es}` override

### PR DESCRIPTION
**Pull request by leemthompo**
cc @colleenmcginnis 

## Caveat lector

I'm 93% sure this didn't fix 100% of everything, but it's probably a fine start.

## Preview

Feast your eyes on https://github.com/elastic/docs-content/pull/196#issuecomment-2476880322 iff CI goes green

### CODE REVIEW

File: serverless/index.asciidoc [hunk 1]
```diff
@@ -11,7 +11,6 @@ include::{asciidoc-dir}/../../shared/attributes.asciidoc[]
 :sec-badge: <<what-is-security-serverless,image:images/sec-badge.svg[Security]>>
 
 // The values of these attributes are different in stateful vs serverless
-:es: Elasticsearch Serverless
```
Comment (leemthompo) [inline serverless/index.asciidoc:14]: This is the important change for Elasticsearch writers, to continue to be able to use ye olde `{es}` for plain olde Elasticsearch

File: serverless/pages/welcome-to-serverless.asciidoc [hunk 2]
```diff
@@ -18,8 +18,8 @@ focus more on gaining value and insight from your data.
 
 Elastic provides three serverless solutions available on {ecloud}:
 
-* **{es}**: Build powerful applications and search experiences using a rich ecosystem of vector search capabilities, APIs, and libraries.
-* **Elastic {observability}**: Monitor your own platforms and services using powerful machine learning and analytics tools with your logs, metrics, traces, and APM data.
+* **{es-serverless}**: Build powerful applications and search experiences using a rich ecosystem of vector search capabilities, APIs, and libraries.
+* **{observability}**: Monitor your own platforms and services using powerful machine learning and analytics tools with your logs, metrics, traces, and APM data.
```
Comment (leemthompo) [inline serverless/pages/welcome-to-serverless.asciidoc:22]: 🚗

File: serverless/pages/developer-tools-troubleshooting.asciidoc [hunk 3]
```diff
@@ -288,7 +288,7 @@ GET /my-index-000001/_settings
 You can update dynamic index settings with the
 https://www.elastic.co/docs/api/doc/elasticsearch-serverless/operation/operation-indices-put-settings[**Update index settings API**].
 Changing dynamic index settings for a data stream
-// <DocLink id="enElasticsearchReferenceModifyDataStreams" section="change-a-dynamic-index-setting-for-a-data-stream">Changing dynamic index settings for a data stream</DocLink>
+// <DocLink id="en{es}ReferenceModifyDataStreams" section="change-a-dynamic-index-setting-for-a-data-stream">Changing dynamic index settings for a data stream</DocLink>
```
Comment (leemthompo) [inline serverless/pages/developer-tools-troubleshooting.asciidoc:291]:
```suggestion
// <DocLink id="enElasticsearchReferenceModifyDataStreams" section="change-a-dynamic-index-setting-for-a-data-stream">Changing dynamic index settings for a data stream</DocLink>
```

**Review by leemthompo — COMMENTED** [review]
False positive

File: serverless/pages/clients-java-getting-started.asciidoc [hunk 4]
```diff
@@ -96,7 +96,7 @@ ElasticsearchTransport transport = new RestClientTransport(
 ElasticsearchClient esClient = new ElasticsearchClient(transport);
 ----
 
-To get API keys or the Elasticsearch Endpoint for a project, see <<elasticsearch-get-started>>.
+To get API keys for the {es} Endpoint for a project, see <<elasticsearch-get-started>>.
```
Comment (lcawl) [inline serverless/pages/clients-java-getting-started.asciidoc:99]: > To get API keys for the {es} Endpoint for a project, see <<elasticsearch-get-started>>.

Not at all the point of this PR, but I don't think endpoint should be capitalized. That'll be something for another cleanup some day.

Comment (leemthompo) [reply serverless/pages/clients-java-getting-started.asciidoc]: I did a lot of drive-by's in this PR, so one more won't hurt

File: serverless/pages/manage-your-project-rest-api.asciidoc [hunk 5]
```diff
@@ -71,7 +71,7 @@ export API_KEY="YOUR_GENERATED_API_KEY"
 
 [discrete]
 [[general-manage-project-with-api-create-a-serverless-elasticsearch-project]]
-=== Create a serverless Elasticsearch project
+=== Create a {es-serverless} project
```
Comment (lcawl) [inline serverless/pages/manage-your-project-rest-api.asciidoc:74]: Does this resolve to "an Elasticsearch Serverless project" now?

```suggestion
=== Create an {es-serverless} project
```

**Review by lcawl — DISMISSED** [review]
I dropped one more comment but otherwise the replacements all make sense to me!

**Review by georgewallace — APPROVED** [review]
Looks good to me. I scoured the pages as well as across Observability and Security and didn't see any places where the variables are incorrect.

**Comment by georgewallace** [comment]
Unfortunately I made the last change so someone else needs to approve now. :(

**Comment by leemthompo** [comment]
>  I scoured the pages as well as across Observability and Security and didn't see any places where the variables are incorrect.

Thanks @georgewallace!

**Review by florent-leborgne — APPROVED** [review]
🌻

**Pull request merged**